docs/application/native/guides/applications/tizen-core.md

```diff
@@ -0,0 +1,392 @@
+    return NULL;
+  }
+
+  ret = tizen_core_poll_fd_set_fd(poll_fd, fd);
+  if (ret != TIZEN_CORE_ERROR_NONE) {
+    dlog_print(DLOG_ERROR, LOG_TAG, "Failed to set fd(%d) to poll fd. error=%d", fd, ret);
+    tizen_core_poll_fd_destroy(poll_fd);
+    return NULL;
+  }
+
+  ret = tizen_core_poll_fd_set_events(poll_fd, events);
+  if (ret != TIZEN_CORE_ERROR_NONE) {
+    dlog_print(DLOG_ERROR, LOG_TAG, "Failed to set request events. error=%d", ret);
+    tizen_core_poll_fd_destroy(poll_fd);
+    return NULL;
+  }
+
+  return poll_fd;
+}
+```
+The created `poll_fd` handle can be registered to a source using `tizen_core_source_add_poll()`.
+When the `poll_fd` is no longer needed, you must call `tizen_core_poll_fd_destroy()`.
+
+2. Creating a source and registering a poll file descriptor
+This example registers the poll fd created in the previous example to a source.
+```c
+static tizen_core_source_h create_fd_source(tizen_core_poll_fd_h poll_fd)
+{
+  tizen_core_source_h source = NULL;
+  tizen_core_h core = NULL;
+  int ret;
+
+  ret = tizen_core_find("task1", &core);
+  if (ret != TIZEN_CORE_ERROR_NONE) {
+    dlog_print(DLOG_ERROR, LOG_TAG, "Failed to find core. error=%d", ret);
+    return NULL;
+  }
+
+  ret = tizen_core_source_create(&source);
+  if (ret != TIZEN_CORE_ERROR_NONE) {
+    dlog_print(DLOG_ERROR, LOG_TAG, "Failed to create source. error=%d", ret);
+    return NULL;
+  }
+
+  ret = tizen_core_source_add_poll(source, poll_fd);
+  if (ret != TIZEN_CORE_ERROR_NONE) {
+    dlog_print(DLOG_ERROR, LOG_TAG, "Failed to add poll fd. error=%d", ret);
+    tizen_core_source_destroy(source);
+    return NULL;
+  }
+
+  ret = tizen_core_add_source(core, source);
+  if (ret != TIZEN_CORE_ERROR_NONE) {
+    dlog_print(DLOG_ERROR, LOG_TAG, "Failed to add source to core. error=%d", ret);
+    tizen_core_source_remove_poll(source, poll_fd);
+    tizen_core_source_destroy(source);
+    return NULL;
+  }
+
+  return source;
+}
+```
+This example registers the poll fd to a source and adds the source to the "task1" core.
+The source begins polling the fd. You should call `tizen_core_remove_source()` to remove the source added with `tizen_core_add_source()` when it is no longer needed.
+
+3. Registering callbacks and controlling events
+In case of file descriptor sources, the `tizen_core_source_prepare_cb()` function usually returns 'false' because we have to wait until poll() gets called before knowing whether or not we should process any events. We set the returned timeout value to -1 indicating we don't care how long the poll() call block lasts. The `tizen_core_source_check_cb()` function tests the result of the poll() call to verify if the required condition is met and returns 'true' if it does.
+This example demonstrates registering `tizen_core_source_prepare_cb()`, `tizen_core_source_check_cb()`, `tizen_core_source_dispatch_cb()`, `tizen_core_source_finalize_cb()` functions and reading data from the fd.
```

Comment:
```suggestion
### Registering callbacks and controlling events
In case of file descriptor sources, the `tizen_core_source_prepare_cb()` function usually returns 'false' because we have to wait until poll() gets called before knowing whether or not we should process any events. We set the returned timeout value to -1 indicating we don't care how long the poll() call block lasts. The `tizen_core_source_check_cb()` function tests the result of the poll() call to verify if the required condition is met and returns 'true' if it does.
This example demonstrates registering `tizen_core_source_prepare_cb()`, `tizen_core_source_check_cb()`, `tizen_core_source_dispatch_cb()`, `tizen_core_source_finalize_cb()` functions and reading data from the fd:
```